File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/MaxMin/MaxMin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleTestSolutions.HackerRank.MaxMin
{
    class MaxMin
    {
        public static int maxMin(int k, List<int> arr)
        {
            int result = Int32.MaxValue;
            arr.Sort();            

            for (int i = 0; i < arr.Count - k + 1; i++)
            {
                result = Math.Min(result, 
                    Math.Max(arr[i], arr[i + k - 1]) - Math.Min(arr[i], arr[i + k - 1]));                
            }
            return result;
        }

    }
}

File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/MaxMin/MaxMin.java
import java.util.Collections;
import java.util.List;

public class MaxMin {
    public static int MaxMin(int k, List<Integer> arr) {
        int result = Integer.MAX_VALUE;
        Collections.sort(arr);

        for (int i = 0; i <= arr.size() - k; ++i)
        {
            result =    Math.min(result,
                    Math.max(arr.get(i), arr.get(i + k - 1)) -
                            Math.min(arr.get(i), arr.get(i + k - 1)));
        }
        return result;
    }
}